nautobot/ipam/api/views.py

```diff
@@ -259,23 +272,9 @@ def available_ips(self, request, pk=None):
                 serializer.save()
                 return Response(serializer.data, status=status.HTTP_201_CREATED)
 
-        # Determine the maximum number of IPs to return
         else:
-            try:
-                limit = int(request.query_params.get("limit", get_settings_or_config("PAGINATE_COUNT")))
-            except ValueError:
-                limit = get_settings_or_config("PAGINATE_COUNT")
-            if get_settings_or_config("MAX_PAGE_SIZE"):
-                limit = min(limit, get_settings_or_config("MAX_PAGE_SIZE"))
-
-            # Calculate available IPs within the prefix
-            ip_list = []
-            for index, ip in enumerate(prefix.get_available_ips(), start=1):
-                ip_list.append(ip)
-                if index == limit:
-                    break
             serializer = serializers.AvailableIPSerializer(
-                ip_list,
+                list(prefix.get_available_ips()),
```

Comment:
This can take an excessive amount of time and memory if the prefix is a large one (e.g. `10.0.0.0/8`) since it is generating a list of *all* available IPs in the prefix. Presumably this is why the pre-existing code did an iterate-and-then-break approach to generating the list instead.